### DISCUSSION TITLE
RangeError when trying to initialise display on my ESP32S3-BOX-LITE devkit

### DISCUSSION THREAD

**Discussion by sambenko** [microsoft/devicescript#649] [Q&A] [answered]
Hello,

I've started working with Devicescript and trying to initialise display for different [esp32s3 box devkits](https://github.com/espressif/esp-box/tree/master). I've created a custom board json file and loaded the pins, configured the spi and created new ST7789Driver instance. However, I can't input the actual display size of my device, the LCD screen has the resolution of 240x320 and I get this error:

```
deploy to NH67
DEV> stop program
  --> done, 10.8kb
DEV> Espressif ESP32-S3-BOX-LITE v2.15.12; file v2.15.15
DEV> start: ./ v0.0.1-3bba539-2023.10.19.15.09
DEV> Unhandled exception
DEV> Exception: RangeError
DEV>  message: invalid dimensions 240x320x4
DEV>  at main [as F0_pc93] (src/main.ts:16:1)
DEV> Unhandled exception
DEV> stop program
```

even when I switched the values 240 and 320, still getting the same error. This is my main.ts (the highest I can go with values is 260 and 220):
```
import { pins } from "@dsboard/esp32s3_box_lite";
import * as ds from "@devicescript/core";
import { ST7789Driver } from "@devicescript/drivers";
import { spi } from "@devicescript/spi";
import { Image } from "@devicescript/graphics";

spi.configure({
    mosi: pins.P6,
    sck: pins.P7,
    miso: pins.P13,
    hz: 40_000_000,
})

pins.P45.setMode(ds.GPIOMode.OutputHigh) // backlight set high

const display = new ST7789Driver(Image.alloc(260, 220, 4), { 
    dc: pins.P4,
    cs: pins.P5,
    reset: pins.P48,
    flip: true, 
    spi: spi,
    offX: 0,
    offY: 0,  
})

pins.P45.setMode(ds.GPIOMode.OutputLow) // backlight set low

await display.init();
await display.image.printCenter("Hello World!", display.image.height / 2);
await display.show();
```

Is my device's display too big for current implementation of the driver? Also, there seems to be no display reset/clear mechanic yet. How would I refresh the display in some other ways? Thank you in advance for looking into this, here is the relevant repository:
https://github.com/sambenko/esp32s3-box-lite-devicescript-display

**Comment by pelikhan** [comment]
@mmoskal

**Comment by mmoskal** [comment]
Unfortunately, we set maximum allocation size to 0xe000; It's possible it could be closer to 0xffff, but either way it's not enough for your screen. The allocation size is set so that we can use uint16_t as size of various buffers, so it's not easy to increase past 0xffff... :/

**Reply by mmoskal** [reply, marked as answer]
Actually, I take that back - we're using 4bpp so the screen should fit. 

Check out esp32 v2.15.1500 when it builds.

**Reply by sambenko** [reply]
Hello, sorry for the late reply, I was sick for a week. I'm not really sure what to do, I checked this https://github.com/microsoft/devicescript-esp32/releases/tag/v2.15.1500, not sure what to go off of next.
Doesn't 4bpp mean that more space needs to be alocated for each pixel? @mmoskal

**Comment by pelikhan** [comment]
Let me push a release and you’ll be able to reflash the ESP32 firmware to get this fix.
________________________________

**Reply by sambenko** [reply]
Thank you, updating to the latest release works for the display size of 320x240 now!

**Comment by pelikhan** [comment]
[celebrate]     Peli de Halleux reacted to your message:
________________________________

**Comment by aeroniemi** [comment]
I've got a similar issue trying to allocate a 320x240x4 image on a pi pico-w - is this a fix that it'll be possible to apply to RP2040?

**Reply by mmoskal** [reply]
I've just bumped devicescript version in the pico port, so firmware v2.15.1900 for pico should have it fixed!

**Comment by pelikhan** [comment]
https://github.com/microsoft/devicescript/pull/663
